src/diffusers/modeling_utils.py

```diff
@@ -296,6 +311,16 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         torch_dtype = kwargs.pop("torch_dtype", None)
         subfolder = kwargs.pop("subfolder", None)
         device_map = kwargs.pop("device_map", None)
+        fast_load = kwargs.pop("fast_load", True)
+
+        # Check if we can handle device_map and dispatching the weights
+        if device_map is not None and not is_torch_version(">=", "1.9.0"):
+            raise NotImplementedError("Loading and dispatching requires torch >= 1.9.0")
+
+        # Fast init is only possible if torch version is >= 1.9.0
+        _INIT_EMPTY_WEIGHTS = fast_load or device_map is not None
+        if _INIT_EMPTY_WEIGHTS and not is_torch_version(">=", "1.9.0"):
+            logger.warn("Loading with `fast_load` requires torch >= 1.9.0. Falling back to normal loading.")
```

Comment:
As discussed offline with @patrickvonplaten added an option to allow disabling fast loading. This is because fast loading won't handle missing or unexpected keys, this use-case might be needed for some fine-tuning experiments for example add some new layers in unet or van and fine-tune. 

The option `fast_load` will be set to `True` by default, but should be disabled if user wants the extra care when loading weights.

Not sure if `fast_load` is the best name, open for suggestions :) 

@patrickvonplaten @anton-l @pcuenca

Reply:
Very cool, I suggested to bypass fast loading for debugging but didn't think about those other use cases. Looks great!

I don't think `fast_load` is too bad. If we want to be more precise, maybe `init_empty`?